File: app/src/main/java/com/example/changfeng/taptapword/WordActivity.java
package com.example.changfeng.taptapword;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import butterknife.Bind;
import butterknife.ButterKnife;
import butterknife.OnClick;


public class WordActivity extends Activity {

    private static final String TAG = "WordActivity";

    public static final String EXTRA_WORD_NAME = "word_name";

    @Bind(R.id.word_name)
    TextView wordNameEditText;

    @Bind(R.id.date_time)
    TextView dateTimeTextView;

    @Bind(R.id.archive)
    CheckBox archiveCheckBox;

    @Bind(R.id.phone_label)
    TextView phoneLabelTextView;

    @Bind(R.id.phone)
    TextView phoneTextView;

    @Bind(R.id.means_label)
    TextView meansLabelTextView;
    @Bind(R.id.means)
    TextView meansTextView;

    @Bind(R.id.web_explains_label)
    TextView webExpainsLabelTextView;
    @Bind(R.id.web_explains)
    TextView webExplainsTextView;

    @Bind(R.id.note_label)
    TextView noteLabelTextView;
    @Bind(R.id.note)
    EditText noteEditText;

    @OnClick(R.id.cancel)
    void quit(View v) {
        finish();
    }

    @OnClick(R.id.save)
    void save(View v) {
        updateWord();
        WordManger.get(this).updateWord(word);
        showToast(getString(R.string.message_save_success));
        finish();
    }

    Word word;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_word);
        ButterKnife.bind(this);

        Intent intent = getIntent();
        String wordName = intent.getStringExtra(EXTRA_WORD_NAME);
        word = WordManger.get(this).getWord(wordName);
        if (word != null) {
            wordNameEditText.setText(word.getName());

            dateTimeTextView.setText(String.format("%04d-%02d-%02d %02d:%02d:%02d", word.getYear(), word.getMonth(), word.getDate(), word.getHour(), word.getMinute(), word.getMinute()));

            archiveCheckBox.setChecked(word.isArchived());

            phoneTextView.setText(word.getFormatPhones());

            meansTextView.setText(word.getMeans());

            webExplainsTextView.setText(word.getWebExplains());

            noteEditText.setText(word.getNote());

            if (word.getFormatPhones().isEmpty()) {
                phoneLabelTextView.setVisibility(View.GONE);
                phoneTextView.setVisibility(View.GONE);
            }

            if (word.getMeans() == null || word.getMeans().isEmpty()) {
                meansLabelTextView.setVisibility(View.GONE);
                meansTextView.setVisibility(View.GONE);
            }

            if (word.getWebExplains() == null || word.getWebExplains().isEmpty()) {
                webExpainsLabelTextView.setVisibility(View.GONE);
                webExplainsTextView.setVisibility(View.GONE);
            }
        }
    }


    private void updateWord() {
        boolean archived = archiveCheckBox.isChecked();
        word.setArchived(archived);
        word.setMeans(meansTextView.getText().toString());
        word.setNote(noteEditText.getText().toString());
    }

    private void showToast(String message) {
        Toast.makeText(this, message, Toast.LENGTH_SHORT).show();
    }
}

File: app/src/main/java/com/example/changfeng/taptapword/WordActivity.kt
package com.example.changfeng.taptapword

import android.app.Activity
import android.os.Bundle
import android.view.View
import android.widget.CheckBox
import android.widget.EditText
import android.widget.TextView
import org.jetbrains.anko.find
import org.jetbrains.anko.onClick
import org.jetbrains.anko.toast


class WordActivity : Activity() {

    val wordNameEditText: TextView get() = find(R.id.word_name)
    val dateTimeTextView: TextView get() = find(R.id.date_time)
    val archiveCheckBox: CheckBox get() = find(R.id.archive)
    val phoneLabelTextView: TextView get() = find(R.id.phone_label)
    val phoneTextView: TextView get() = find(R.id.phone)
    val meansLabelTextView: TextView get() = find(R.id.means_label)
    val meansTextView: TextView get() = find(R.id.means)
    val webExplainsLabelTextView: TextView get() = find(R.id.web_explains_label)
    val webExplainsTextView: TextView get() = find(R.id.web_explains)
    val noteEditText: EditText get() = find(R.id.note)


    var word: Word? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_word)

        find<TextView>(R.id.save).onClick {
            updateWord()
            WordManger.get(this).updateWord(word)
            toast(R.string.message_save_success)
            finish()
        }

        find<TextView>(R.id.cancel).onClick {
            finish()
        }

        val wordName = intent.getStringExtra(EXTRA_WORD_NAME)
        word = WordManger.get(this).getWord(wordName)

        val w = word

        if (w != null) {
            wordNameEditText.text = w.name
            dateTimeTextView.text = "${w.year}-${w.month}-${w.date} ${w.hour}:${w.minute}:${w.second}"
            archiveCheckBox.isChecked = w.isArchived
            phoneTextView.text = w.formatPhones
            meansTextView.text = w.means
            webExplainsTextView.text = w.webExplains
            noteEditText.setText(w.note)

            if (w.formatPhones.isNullOrEmpty()) {
                phoneLabelTextView.visibility = View.GONE
                phoneTextView.visibility = View.GONE
            }

            if (w.means.isNullOrEmpty()){
                meansLabelTextView.visibility = View.GONE
                meansTextView.visibility = View.GONE
            }

            if (w.webExplains.isNullOrEmpty()) {
                webExplainsLabelTextView.visibility = View.GONE
                webExplainsTextView.visibility = View.GONE
            }

        }
    }


    private fun updateWord() {
        word?.isArchived = archiveCheckBox.isChecked
        word?.means = meansTextView.text.toString()
        word?.note = noteEditText.text.toString()
    }


    companion object {

        private val TAG = "WordActivity"

        val EXTRA_WORD_NAME = "word_name"
    }
}
